app.py

```diff
@@ -0,0 +1,70 @@
+import traceback
+import json
+import requests
+
+from kafka import KafkaConsumer
+
+from kerlescan.inventory_service_interface import fetch_systems_with_profiles
+
+from system_profile_archiver import logging, metrics, config
+
+logger = logging.initialize_logging()
+
+
+def main():
+    logger.info("starting listener")
+
+    consumer = init_consumer()
+
+    counters = {
+        "inventory_service_requests": metrics.inventory_service_requests,
+        "inventory_service_exceptions": metrics.inventory_service_exceptions,
+        "systems_compared_no_sysprofile": metrics.inventory_service_no_profile,
+    }
+
+    while True:
+        for data in consumer:
+            logger.info("consuming message")
+            try:
+                inventory_uuid = data.value["host"]["id"]
+                identity = data.value["platform_metadata"]["b64_identity"]
+                systems = fetch_systems_with_profiles(
+                    [inventory_uuid], identity, logger, counters
+                )
+                system = systems[0]["system_profile"]
+                system["system_profile_exists"] = True
+                system["display_name"] = systems[0]["display_name"]
+                profile = {"inventory_id": inventory_uuid, "profile": system}
+                headers = {
+                    "x-rh-identity": identity,
+                    "content-type": "application/json",
+                }
+                result = requests.post(
```

Comment:
Could we break this out to its own method or put a blank line above to help separate the logic ? I'm also wondering if that would make testing easier.